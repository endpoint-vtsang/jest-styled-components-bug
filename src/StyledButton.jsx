import React from 'react';
import styled from 'styled-components';

export const StyledButton = styled('button')`
  color: white;
  padding: 40px;
  background: black;

  &[data-focus-visible-added]:not(:disabled),
  &:focus-visible:not(:disabled) {
    background: red;
    box-shadow: green;
  }
`;
