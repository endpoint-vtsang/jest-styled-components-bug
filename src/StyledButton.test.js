import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { StyledButton } from "./StyledButton";

describe("Button", () => {
  it("renders the component", () => {
    const { container } = render(<StyledButton />);

    expect(container).toMatchSnapshot();
  });

  it("renders a focus ring", async () => {
    render(<StyledButton />);

    const myButton = screen.getByRole("button");

    await userEvent.tab();

    await expect(myButton).toHaveFocus();
    await expect(myButton).toHaveStyleRule("box-shadow", "green", {
      modifier: "&:focus-visible:not(:disabled)",
    });
  });
});
