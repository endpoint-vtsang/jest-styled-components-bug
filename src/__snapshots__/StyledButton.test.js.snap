// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders the component 1`] = `
.c0 {
  color: white;
  padding: 40px;
  background: black;
}

.c0[data-focus-visible-added]:not(:disabled),.c0:focus-visible:not(:disabled) {
  background: red;
  box-shadow: green;
}

<div>
  <button
    class="c0"
  />
</div>
`;
